src/mmw/js/src/modeling/controllers.js

```diff
@@ -9,66 +9,58 @@ var $ = require('jquery'),
     models = require('./models');
 
 var ModelingController = {
-    modelPrepare: function() {
-        if (!App.map.get('areaOfInterest')) {
+    modelPrepare: function(projectId) {
+        if (!projectId && !App.map.get('areaOfInterest')) {
             router.navigate('', { trigger: true });
             return false;
         }
     },
 
-    model: function() {
-        // TODO: If you are coming to the page with a
-        // project ID in the URL, we need to load an
-        // existing  project here.
-        // i.e. var project = new models.ProjectModel({
-        //  id: XX
-        //  }).fetch
-        //  .done(init everything else);
+    model: function(projectId) {
+        var project;
 
-        // TODO: For a new project, users will eventually
-        // be able to choose which modeling package
-        // they want to use in their project. For
-        // now, the only option is TR55, so it is
-        // hard-coded here.
+        if (projectId) {
+            project = new models.ProjectModel({
+                id: projectId
+            });
+
+            project
+                .fetch()
+                .done(function(data) {
+                    App.map.set('areaOfInterest', project.get('area_of_interest'));
+                    initViews(project);
+                });
+        } else {
+            // The structure of the AoI is slightly different depending
+            // on whether the shape was drawn or selected.
+            var aoi = App.map.get('areaOfInterest').geometry ?
+                      App.map.get('areaOfInterest').geometry :
+                      App.map.get('areaOfInterest');
```

Comment:
I don't know if this has any impact on this, but I recently modified the GeoJSON that gets populated for selected shapes (https://github.com/WikiWatershed/model-my-watershed/commit/ce884f6e5b3fe9dc1453c1d2fc8f535449aa3223#diff-059395c30679e53cc84d53e77c16f93dL107).